sktime/forecasting/var_reduce.py

```diff
@@ -0,0 +1,296 @@
+
+        # Initialize placeholders for coefficients and intercepts
+        coefficients = np.zeros((self.lags, num_series, num_series))
+        intercepts = np.zeros(num_series)
+
+        # Fit the chosen regressor model for each series
+        for i, var_name in enumerate(var_names):
+            model = deepcopy(self.regressor_)
+            y = Y[var_name]
+            model.fit(X, y)
+            self.regressors[var_name] = model
+
+            # if the model has `.intercept_` and `.coef_` attributes, extract them
+            if hasattr(model, "intercept_") and hasattr(model, "coef_"):
+                intercepts[i] = model.intercept_
+                # Reshape the coefficients to match with statsmodels VAR
+                coefficients[:, :, i] = model.coef_.reshape(self.lags, num_series)
+            else:
+                pass
+
+        # if the model has `.intercept_` and `.coef_` attributes, extract them
+        if hasattr(model, "intercept_") and hasattr(model, "coef_"):
+            # Transpose coefficients to match the order of statsmodels' VAR
+            self.coefficients_ = np.transpose(coefficients, (0, 2, 1))
+            self.intercept_ = intercepts
+        else:
+            self.coefficients_ = None
+            self.intercept_ = None
+        return self
+
+    def _predict(self, fh, X=None):
+        """Forecast time series at future horizon.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            The forecasters horizon with the steps ahead to to predict.
+            Default is one-step ahead forecast,
+            i.e. np.array([1])
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored.
+
+        Returns
+        -------
+        y_pred : pd.DataFrame
+            Series of predicted values
+        """
+        # Get the last available values for prediction
+        y_last = self._y.iloc[-self.lags :]
+
+        # Initialize a list to store predictions
+        y_pred = []
+
+        for _ in range(1, len(fh) + 1):
+            # Prepare X
+            X_last = self.prepare_for_predict(y_last, return_as_ndarray=False)
+
+            # Make timestep specific predictions for each variable using its regressor
+            y_pred_step = []
+            for var_name in self.var_names:
+                model = self.regressors[var_name]
+                y_pred_step_var = model.predict(X_last).item()  # is a float
+                y_pred_step.append(y_pred_step_var)
+
+            # Convert y_pred_step into a one-row DataFrame
+            y_pred_step = pd.DataFrame([y_pred_step], columns=self.var_names)
+            y_pred.append(y_pred_step)
+
+            # Append the new predictions (y_pred_step) at the end of y_last
+            y_last = pd.concat([y_last, y_pred_step], axis=0)
+
+        # Concatenate the list of Series into the final DataFrame
+        y_pred = pd.concat(y_pred)
+        y_pred.index = fh.to_absolute_index()
```

Comment:
either you need to pass the `self.cutoff` as an arg, or use the methog `get_expeced_pred_idx` (which also wants the cutoff)

Reply:
(Addressed and resolved)